### PR TITLE
Completed TDD Rspec Palindrome homework #2

### PR DESCRIPTION
**Pull request by daktest**
- Added tests for the Television class

[#152207328]

### CODE REVIEW

File: objects_tdd/spec/lib/television_spec.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      subject.power
+      expect(subject.on?).to eq(true)
+      subject.power
+      expect(subject.on?).to eq(false)
+    end
+  end
+
+  describe '#set_channel(number)' do
+    it 'sets the channel to the number if within valid range' do
+      subject.power
```
Comment (jasonnoble) [inline objects_tdd/spec/lib/television_spec.rb]: If you need to turn the power on before changing the channel, that's a good use for a `before` block:

    before(:each) do
      subject.power
    end

File: objects_tdd/spec/lib/television_spec.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      expect(subject.channel).to eq(3)
+    end
+  end
+
+  describe '#channel_up' do
+    it 'increments the channel by one' do
+      subject.power
+      subject.channel_up
+      expect(subject.channel).to eq(4)
+    end
```
Comment (jasonnoble) [inline objects_tdd/spec/lib/television_spec.rb]: What if you hit `channel_up` and you're already on the highest channel?  Write a test that verifies it "rolls" over to the lowest channel.

File: objects_tdd/spec/lib/television_spec.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  end
+
+  describe '#channel_down' do
+    it 'decrements the channel by one' do
+      subject.power
+      subject.channel_set(37)
+      subject.channel_down
+      expect(subject.channel).to eq(36)
```
Comment (jasonnoble) [inline objects_tdd/spec/lib/television_spec.rb]: Same comment for lowest channel should "roll" over to the highest channel.

File: objects_tdd/spec/lib/television_spec.rb [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    end
+  end
+
+  describe 'channel loops when end of range is reached' do
```
Comment (jasonnoble) [inline objects_tdd/spec/lib/television_spec.rb]: This should be inside the `#channel_up` block

File: objects_tdd/spec/lib/television_spec.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    it '#channel_up loops to first channel after reaching last channel' do
+      subject.power
+      subject.channel_set(500)
+      subject.channel_up
+      expect(subject.channel).to eq(3)
+    end
+
+    it '#channel_down loops to last channel after reaching first channel' do
```
Comment (jasonnoble) [inline objects_tdd/spec/lib/television_spec.rb]: Inside the `#channel_down` block.

**Review by jasonnoble — CHANGES_REQUESTED** [review]

File: objects_tdd/spec/lib/television_spec.rb (hunk 1 above)
Comment (daktest) [reply objects_tdd/spec/lib/television_spec.rb]: @jasonnoble There is one test that should not have the power turned on.  How do you exclude a single test from running the before block?

**Pull request merged**